corehq/apps/data_dictionary/views.py

```diff
@@ -143,6 +149,144 @@ def data_dictionary_json(request, domain, case_type_name=None):
     })
 
 
+@login_and_domain_required
+@requires_privilege_with_fallback(privileges.DATA_DICTIONARY)
+def data_dictionary_json_case_types(request, domain):
+    fhir_resource_type_name_by_case_type = {}
+    if toggles.FHIR_INTEGRATION.enabled(domain):
+        fhir_resource_type_name_by_case_type = load_fhir_case_type_mapping(domain)
+
+    case_type_app_module_count = get_case_type_app_module_count(domain)
+    used_props_by_case_type = get_used_props_by_case_type(domain)
+    geo_case_prop = get_geo_case_property(domain)
+
+    queryset = CaseType.objects.filter(domain=domain).annotate(properties_count=Count('property'))
+    if not request.GET.get("load_deprecated_case_types", False) == "true":
+        queryset = queryset.filter(is_deprecated=False)
+    case_types_data = []
+    for case_type in queryset:
+        used_props = used_props_by_case_type.get(case_type.name, [])
+        case_types_data.append({
+            "name": case_type.name,
+            "fhir_resource_type": fhir_resource_type_name_by_case_type.get(case_type),
+            "is_deprecated": case_type.is_deprecated,
+            "module_count": case_type_app_module_count.get(case_type.name, 0),
+            "is_safe_to_delete": len(used_props) == 0,
+            "properties_count": case_type.properties_count,
+        })
+
+    return JsonResponse({
+        "case_types": case_types_data,
+        "geo_case_property": geo_case_prop,
+    })
+
+
+@login_and_domain_required
+@requires_privilege_with_fallback(privileges.DATA_DICTIONARY)
+def data_dictionary_json_case_properties(request, domain, case_type_name):
+    fhir_resource_type_name_by_case_type = {}
+    fhir_resource_prop_by_case_prop = {}
+    if toggles.FHIR_INTEGRATION.enabled(domain):
+        fhir_resource_type_name_by_case_type = load_fhir_case_type_mapping(domain)
+        fhir_resource_prop_by_case_prop = load_fhir_case_properties_mapping(domain)
+
+    case_type_app_module_count = get_case_type_app_module_count(domain)
+    used_props_by_case_type = get_used_props_by_case_type(domain)
+    geo_case_prop = get_geo_case_property(domain)
+
+    def _get_case_data(case_type):
```

Comment:
If you pull this function out, then you can avoid the duplicated code in `data_dictionary_json_case_types()` view.

In case it's helpful, when I was thinking about this in code, mine looked like this:

```python
def get_case_type_data(domain, case_type):
    case_type_app_module_count = get_case_type_app_module_count(domain)
    used_props_by_case_type = get_used_props_by_case_type(domain)
    module_count = case_type_app_module_count.get(case_type.name, 0)
    used_props = used_props_by_case_type.get(case_type.name, [])
    data = {
        "name": case_type.name,
        "fhir_resource_type": None,
        "is_deprecated": case_type.is_deprecated,
        "module_count": module_count,
        "is_safe_to_delete": len(used_props) == 0,
        "properties_count": case_type.properties_count,
    }
    if toggles.FHIR_INTEGRATION.enabled(domain):
        fhir_resource_type_name_by_case_type = load_fhir_case_type_mapping(domain)
        data["fhir_resource_type"] = fhir_resource_type_name_by_case_type.get(case_type)
    return data
```

You should decorate `get_case_type_app_module_count()`  with `@quickcache(vary_on=['domain'], timeout=60 * 10)`.